src/metabase/driver/sql_jdbc/sync/describe_table.clj

```diff
@@ -255,16 +269,29 @@
    java.lang.Double                :type/Float
    java.lang.Number                :type/Number
    java.lang.Boolean               :type/Boolean
+   java.time.LocalDateTime         :type/DateTime
    clojure.lang.PersistentVector   :type/Array
    clojure.lang.PersistentArrayMap :type/Structured})
 
+(def ^:const db-type-map
```

Comment:
This doesn't really need to be `^:const`. 

When something is `^:const` 
- It will be evaluated at compile time vs launch. Handy in some places if we want to say look at directory structure of the source code or do some expensive calculation when running locally without AOT but want that to be pre-calculated when we build the uberjar and AOT everything. Here, it makes zero difference
-  The compiler will splice this inline to places that reference it. This means: 
   - It can't be mocked with `with-redefs` in tests. There's a nonzero chance we'd want to mock this in tests in the future. 
   - Changes to it won't be picked up by things referencing it unless their definitions are reloaded as well. This makes it less useful in the REPL 
   - The entire value will get spliced into every place referencing it. No difference here since you're using it once but if we used it in a few places then it would end up making the code larger

I know we have a lot of other stuff marked `^:const` but IMO most of that stuff is incorrectly marked. Only stuff where we want to do expensive calculations at compile time or need it to be inlined for tight loop/performance purposes should be marked `:^const` IMO. Feel free to un-const stuff that is inappropriately const

Reply:
I added a section on this to the Metabase Clojure Style Guide https://github.com/metabase/metabase/wiki/Metabase-Clojure-Style-Guide#using-const -- feel free to post that link if you're reviewing someone else's PR and come across it